docs/development/languages/how-to-write-go-code.md

```diff
@@ -38,3 +38,54 @@ func DoSomething() (string, error) {
 	return "", fmt.Errorf("%w: %s", ErrUpstreamFailure, fmt.Errorf("upstream error"))
 }
 ```
+
+## Object Construction
+
+When creating constructors for objects (structs) in Go, prefer the **Functional Options Pattern**. This approach uses variadic arguments to handle optional configuration while keeping the API clean and extensible.
+
+*   **Required Arguments:** Pass required dependencies as explicit, typed arguments to the constructor function (e.g., `NewServer`).
+*   **Optional Arguments:** Use variadic arguments (e.g., `...Option`) for optional configuration.
+*   **Defaults:** The struct should be initialized with sane defaults for any optional arguments that are not provided.
+
+### Example
+
+```go
+package server
+
+import (
+	"time"
+)
+
+type Server struct {
+	host    string
+	port    int
+	timeout time.Duration
+}
+
+// Option defines a functional option for configuring the Server.
+type Option func(*Server)
```

Comment:
With the constructor functions, return an error, and then check that error during the construction of the object. 

This allows things like validations to take place or other errors that can only be determined as the function is being applied.